lightpath/path.py

```diff
@@ -253,7 +288,9 @@ def clear(self, wait=False, timeout=None,
         logger.info('Removing devices along the beampath ...')
         status = [device.remove(timeout=timeout)
                   for device in target_devices
-                  if not device.removed and hasattr(device, 'remove')]
+                  if find_device_state(device) in (DeviceState.Inserted,
+                                                   DeviceState.Unknown)
+                  and hasattr(device, 'remove')]
```

Comment:
I don't like four-line list comprehensions for lack of readability, but that's more a matter of personal taste. This seems to do the job.